app/models/job.rb

```diff
@@ -12,39 +15,81 @@ class Job < ApplicationRecord
 
   validates :status, presence: true, inclusion: { in: STATUSES }
   validates :kind, presence: true, inclusion: { in: KINDS }
+  validates :end_time, comparison: { greater_than: :start_time }, if: ->{ end_time.present? }
 
   STATUSES.each do |status_name|
+    # A helper method to determine the status of a job
+    # 
+    # @example job.queued? #=> true
+    # @return [true, false] depending on the status of the job
     define_method("#{status_name}?".to_sym) do
       status == status_name
     end
 
+    # A helper method to update the status of the job
+    #
+    # @example job.mark_as_cancelled #=> true
+    # @return [true, false] depending on success of update method
     define_method("mark_as_#{status_name}".to_sym) do
       update(status: status_name)
     end
   end
 
   KINDS.each do |kind_name|
+    # A helper method to determine the kind of job
+    #
+    # @example job.full? #=> true
+    # @return [true, false] depending on kind of job
     define_method("#{kind_name}?".to_sym) do
       kind == kind_name
     end
   end
 
+  # Returns the fullpath to the extraction folder for this job
+  #
+  # @example job.extraction_folder #=> /app/extractions/development/2023-04-28_08-51-16_-_19
+  # @return String
   def extraction_folder
-    "#{EXTRACTIONS_FOLDER}/#{created_at.strftime('%Y-%m-%d_%H-%M-%S')}_-_#{id}"
+    "#{EXTRACTIONS_FOLDER}/#{created_at.to_fs(:file_format)}_-_#{id}"
   end
 
+  # Creates a folder at the location of the extraction_folder
+  #
+  # @return [true, false] depending on success of the folder creation
   def create_folder
     Dir.mkdir(extraction_folder)
   end
 
+  # Deletes a folder at the location of the extraction folder
+  #
+  # @return [true, false] depending on success of the folder deletion
   def delete_folder
-    return unless File.exist?(extraction_folder)
+    return unless Dir.exist?(extraction_folder)
 
     FileUtils.rm Dir.glob("#{extraction_folder}/*")
     Dir.rmdir(extraction_folder)
   end
 
+  # Converts the files stored in the extraction folder into pageable objects
+  #
+  # @return Extraction::Documents object
   def documents
     Extraction::Documents.new(extraction_folder)
   end
+
+  # Returns the number of seconds a job has been running for
+  #
+  # @return Integer
+  def duration_seconds
+    return if start_time.blank? || end_time.blank?
+
+    end_time - start_time
+  end
+
+  # Returns the size of the extraction folder in bytes 
+  #
+  # @return Integer
+  def extraction_folder_size_in_bytes
+    `du #{extraction_folder} | cut -f1`.to_i
```

Comment:
i guess we need to be careful to not execute unexpected thigs in the extraction folder variable `du /path | rm -rf /`

Reply:
not sure if this is an input that comes from the frontend though, it might be ok?

Reply:
Yeah they can't define the extraction folder, it's worked out by the system :)